src/store.js

```diff
@@ -0,0 +1,298 @@
+      mfor = '#.##0'; //in case of undefined
+    } else {
+      if (measureInfo.qNumFormat.qType == 'R') {
+        mfor = measureInfo.qNumFormat.qFmt;
+        mfor = mfor.replace(/(|)/gi, '');
+      } else {
+        mfor = measureInfo.qNumFormat.qFmt;
+      }
+    }
+
+    MeasuresFormat.push(mfor);
+
+    switch (mfor.substr(mfor.length - 1).toLowerCase()) {
+      case 'm':
+        vExtraLabel = ' (M)';
+        ExtraLabelsArray.push(' (M)');
+        break;
+      case 'k':
+        vExtraLabel = ' (k)';
+        ExtraLabelsArray.push(' (k)');
+        break;
+      default:
+        vExtraLabel = '';
+        ExtraLabelsArray.push('');
+        break;
+    }
+    vNumMeasures++;
+  });
+
+  component.backendApi.eachDataRow(function (t, a) {
+    lastrow = t;
+
+    var vNumMeasuresPlus = vNumMeasures + 1;
+
+    ConceptMatrix[t] = new Array();
+    ConceptMatrix[t][0] = a[0].qText;
+
+    ConceptMatrixFirst[t] = a[0].qText;
+    ConceptMatrixRowElem[t] = a[0].qElemNumber;
+    var nMeasures = 0;
+    if (vNumDims == 1) {
+      for (nMeasures = 1; nMeasures <= vNumMeasures; nMeasures++) {
+        ConceptMatrix[t][nMeasures] = a[nMeasures].qNum;
+      }
+    } else {
+      ConceptMatrix[t][1] = a[1].qText;
+      ConceptMatrixColElem[t] = a[1].qElemNumber;
+      ConceptMatrixSecond[t] = a[1].qText;
+      // set the hipercube in a plain array without pivoting
+      for (nMeasures = 2; nMeasures <= vNumMeasuresPlus; nMeasures++) {
+        ConceptMatrix[t][nMeasures] = a[nMeasures].qNum;
+      }
+    }
+  });
+
+  if (nRows >= (vMaxLoops * 1000)) {
+    alert(vErrorMessage);
+  }
+
+  if (vNumDims == 2) {
+    //new array with unique values for 2nd dim
+    var SecondHeader = ConceptMatrixSecond.filter(onlyUnique);//second dimension concepts
+    ConceptMatrixRowElem = ConceptMatrixRowElem.filter(onlyUnique);//first dimension concepts
+    ConceptMatrixColElem = ConceptMatrixColElem.filter(onlyUnique);//dimension code for further selections
+    var eo = ConceptMatrixColElem.length;
+    var vLoopColsMeasures = 1;
+    ConceptMatrixColElemTable[0] = ConceptMatrixColElem[0];
+    for (var xx = 0; xx < eo; xx++) {
+      if (vSeparatorCols && xx > 0) {
+        ConceptMatrixColElemTable[vLoopColsMeasures] = ConceptMatrixColElem[xx];
+        vLoopColsMeasures++;
+      }
+
+      for (var xxx = 0; xxx < vNumMeasures; xxx++) {
+        ConceptMatrixColElemTable[vLoopColsMeasures] = ConceptMatrixColElem[xx];
+        vLoopColsMeasures++;
+      }
+    }
+
+    ConceptMatrixFirstClean = ConceptMatrixFirst.filter(onlyUnique);
+    SecondHeaderLength = SecondHeader.length;
+    vNumMeasures2 = vNumMeasures * SecondHeaderLength;
+  }
+
+  var ConceptPos = 0;
+  var nMeas3 = 0;
+  var vHeaderIndex = 0;
+  var MeasurePos = 0;
+  for (var nPivotElems = 0; nPivotElems <= lastrow; nPivotElems++) {
+    ConceptMatrixPivot[nPivotElems] = new Array();
+    ConceptPos = ConceptMatrixFirstClean.indexOf(ConceptMatrix[nPivotElems][0]);
```

Comment:
If I start from scratch `ConceptMatrixFirstClean ` is an empty array, so this will always return -1. I guess we miss to initialise the `ConceptMatrixFirstClean` if we want to have a lookup like this. Otherwise looking into `ConecptMatrix` instead? Only guessing. 😧